packages/ember-template-compiler/lib/system/compile.ts

```diff
@@ -33,9 +36,17 @@ export default function compile(templateString: string, options: CompileOptions
     );
   }
 
-  return template(evaluate(precompile(templateString, options)));
-}
+  options.locals = options.locals ?? Object.keys(scopeValues ?? {});
+  let [block, usedLocals] = precompileJSON(templateSource, compileOptions(options));
+  let reifiedScopeValues = usedLocals.map((key) => scopeValues[key]);
+
+  let templateBlock: SerializedTemplateWithLazyBlock = {
+    id: String(templateId++),
+    block: JSON.stringify(block),
+    moduleName: options.moduleName ?? options.meta?.moduleName ?? '(unknown template module)',
+    scope: reifiedScopeValues.length > 0 ? () => reifiedScopeValues : null,
+    isStrictMode: options.strictMode ?? false,
+  };
```

Comment:
I wonder if we should do this in https://github.com/emberjs/ember.js/blob/8dae0cab0a18e62117347d619b69fb2f3a4ac4f2/packages/%40ember/-internals/glimmer/tests/utils/helpers.js#L1 (since it seems this is mostly used by our own tests)

Instead of changing the external facing public API here. 🤔